Don't render top nav until logged in

TW80000 opened the pull request:
Currently the top nav loads even when the user isn't logged in. This
means that the initE method for child views of the top navigation run
before user is set in the context. This makes it so that the views in
the top nav don't load until the user is properly set in the context.

Required by https://github.com/nanoPayinc/NANOPAY/pull/3492

Commit:
Don't render top nav until logged in

Currently the top nav loads even when the user isn't logged in. This
means that the initE method for child views of the top navigation run
before user is set in the context. This makes it so that the views in
the top nav don't load until the user is properly set in the context.

kgrgreer commented:
reviewing...

zahnno commented on src/foam/nanos/u2/navigation/TopNavigation.js:
```diff
@@ -101,24 +101,33 @@ foam.CLASS({
       this
         .addClass(this.myClass())
+        .start('div', null, this.welcome$)
+          .add('Welcome').addClass('welcome-label').hide(this.loginSuccess$)
```
Let's make this 'Welcome' string into a messages prop.

TW80000 replied:
Done.

Commit:
Move hard-coded string to messages array

Commit:
Remove unnecessary code

Hiding that part of the view on loginSuccess makes removing the elements
redundant.

TW80000 commented:
All comments addressed. Waiting on approval or new comments.

kgrgreer commented on src/foam/nanos/u2/navigation/TopNavigation.js:
```diff
@@ -97,6 +97,13 @@ foam.CLASS({
 
+  messages: [
+    {
+      name: 'Greeting',
```
Should this be GREETING (or WELCOME). But don't we name messages like constants?

zahnno replied:
@kgrgreer If we're following suit to our previous messages then no, we use PascalCase.

TW80000 replied:
I looked through the project for how we use messages and we do it both ways in many different places.

Commit:
Make message name uppercase

zahnno commented:
@kgrgreer Please address
